dashboard/di_graph.bal

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+
+public type Node record {|
+    readonly string V;
+    string[] E = [];
+    int level = 1;
+|};
+
+class DiGraph {
+
+    private table<Node> key(V) graph = table [];
+
+    public function init() {
+    }
+
+    public function addNode(string node) {
+        if !self.graph.hasKey(node) {
+            self.graph.add({V: node});
+        }
+    }
+
+    public function addEdge(string v1, string v2) {
+        if !self.graph.hasKey(v1) {
+            io:println(`vertex ${v1} doesn't exists`);
+        }
+        else if !self.graph.hasKey(v2) {
+            io:println(`vertex ${v2} doesn't exists`);
+        }
+        else {
+            Node n = self.graph.get(v1);
+            n.E.push(v2);
+        }
+    }
+
+    public function inDegree(string v) returns int {
+        int count = 0;
+        foreach Node n in self.graph {
+            if n.E.indexOf(v) is int {
+                count += 1;
+            }
+        }
+        return count;
+    }
+
+    public function successor(string v) returns string[] {
+        return self.graph.get(v).E;
```

Comment:
What will happen if the `v` does not exist here? (Here too, let's use more meaningful name instead of `v`).